purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -379,11 +379,11 @@ class Purchases internal constructor(
     }
 
     /**
-     * Make a purchase upgrading from a previous sku. If purchasing a subscription,
+     * Make a purchase upgrading from a previous subscription. If purchasing a subscription,
```

Comment:
I'm confused why we have this "if purchasing a subscription" here, because i didn't think you could upgrade an OTP? 

@tonidero was this possibly just copy pasta when we updated docs to take the PurchaseOptions into account?

Reply:
It was kinda copy paste, however, there is nothing impeding devs from calling this method with an inapp product though right? With the current implementation in `bc5-support`, the `UpgradeInfo` would be ignored, so I guess we can mention that this method should only be used for subscriptions and if using an inapp product `upgradeInfo` will be ignored or something like that.

Reply:
I think we can also check whether the currently purchased product is autorenewing. If it isn't I'd throw at least a warning (and in the future throw an exception)

Reply:
actually i think the purchase will just fail if called with an OTP. we still create a ReplaceProductInfo, meaning we won't just ignore the values for OTP... i'll test it

Reply:
okay, so, i did some debugging on this.

right now, upgrading from sub -> OTP will fail. 

our flow is: [we call `findPurchaseInPurchaseHistory` for the _type_ of the _new product_ to be purchased](https://github.com/RevenueCat/purchases-android/pull/723/files#diff-e019410c58657f001e6ba8be4f066c84c928f4e017ae3b3b6366cebfd64b9716R1563), and then filter for a purchase with the product id of the _old product_.  we then use the purchase to get the purchase token to pass to BillingClient.

for instance, in an upgrade from sub with id `onemonth_freetrial `-> OTP with id `consumable`, we'd call `findPurchaseInPurchaseHistory` for the type `INAPP`, and then try to find a purchase for `onemonth_freetrial` in it. 

this logic is obviously flawed -- we probably should never query for `INAPP` type in the first place, since you can't upgrade to an `INAPP`. but if we are going to pass a type, it should be the type of the old product, because that's the one whose purchase we are looking for.  i'm going to make that update in this PR.

after making that change^^, i tested, and @tonidero was correct that we just drop the UpgradeInfo and go through with a normal OTP flow.

now we have a choice of how to handle in the future... at the very least, we should add an error log. i think the "right" thing would be to error the purchase, since it's not going to perform any sort of upgrade. the user might be confused why the old subscription is still in effect. however, that'd be a behavior change from the last version...

Reply:
Ah makes sense thanks! I think failing the purchase when devs try to do some invalid stuff like upgrading from/to an inapp product makes the most sense. As for a behavior change, I don't think it's ideal, but if we do this change as part of the major and we mention it as part of the migration documentation, I think it should be ok...